core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -135,57 +148,100 @@ case class FeatureDistribution
   }
 
   override def toString(): String = {
-    s"Name=$name, Key=$key, Count=$count, Nulls=$nulls, Histogram=${distribution.toList}, BinInfo=${summaryInfo.toList}"
+    val valStr = Seq(
+      "type" -> `type`.toString,
+      "name" -> name,
+      "key" -> key,
+      "count" -> count.toString,
+      "nulls" -> nulls.toString,
+      "distribution" -> distribution.mkString("[", ",", "]"),
+      "summaryInfo" -> summaryInfo.mkString("[", ",", "]")
+    ).map { case (n, v) => s"$n = $v" }.mkString(", ")
+
+    s"${getClass.getSimpleName}($valStr)"
+  }
+
+  override def equals(that: Any): Boolean = that match {
+    case FeatureDistribution(`name`, `key`, `count`, `nulls`, d, s, `type`) =>
+      distribution.deep == d.deep && summaryInfo.deep == s.deep
+    case _ => false
   }
+
+  override def hashCode(): Int = Objects.hashCode(name, key, count, nulls, distribution, summaryInfo, `type`)
 }
 
-private[op] object FeatureDistribution {
+object FeatureDistribution {
 
   val MaxBins = 100000
 
   implicit val semigroup: Semigroup[FeatureDistribution] = new Semigroup[FeatureDistribution] {
-    override def plus(l: FeatureDistribution, r: FeatureDistribution) = l.reduce(r)
+    override def plus(l: FeatureDistribution, r: FeatureDistribution): FeatureDistribution = l.reduce(r)
+  }
+
+  implicit val formats: Formats = DefaultFormats +
+    EnumEntrySerializer.json4s[FeatureDistributionType](FeatureDistributionType)
```

Comment:
So we have a bunch of these sitting in https://github.com/salesforce/TransmogrifAI/blob/master/features/src/main/scala/com/salesforce/op/stages/OpPipelineStageReadWriteShared.scala#L79-L86

Should we we add this one there instead? Or put the others in their corresponding companion objects?